prometheus_xmpp/__main__.py

```diff
@@ -179,14 +232,125 @@ def message(self, msg):
                 response = "Unknown command: %s" % args[0].lower()
             msg.reply(response).send()
 
+    async def dispatch_message(self, mto, mbody, mhtml):
+        if not self.muc and not self.omemo:
+            self.send_message(mto=mto, mbody=mbody, mhtml=mhtml, mtype="chat")
+        elif not self.muc and self.omemo:
+            await self.send_encrypted(mto=[JID(mto)], mtype="chat", body=mbody)
+        elif self.muc and not self.omemo:
+            self.send_message(
+                mto=self.muc_jid, mbody=mbody, mhtml=mhtml, mtype="groupchat"
+            )
+        elif self.muc and self.omemo:
+            owners, admins, members, outcasts = await asyncio.gather(
+                self.plugin["xep_0045"].get_affiliation_list(self.muc_jid, "owner"),
+                self.plugin["xep_0045"].get_affiliation_list(self.muc_jid, "admin"),
+                self.plugin["xep_0045"].get_affiliation_list(self.muc_jid, "member"),
+                self.plugin["xep_0045"].get_affiliation_list(self.muc_jid, "outcast"),
+                return_exceptions=True,
+            )
+            muc_affiliation = owners + admins + members + outcasts
+            recipients = [JID(nick) for nick in muc_affiliation]
+            recipients.remove(self.jid)
+            logging.debug("sending encrypted message to recipients %s", recipients)
+            await self.send_encrypted(mto=recipients, mtype="groupchat", body=mbody)
+
+    async def send_encrypted(self, mto: JID, mtype: Literal['chat', 'error', 'groupchat', 'headline', 'normal'], body):
+        """Helper to reply with encrypted messages"""
+        if self.muc:
```

Comment:
I have trouble reading this function. It seems like it just ignores mto if self.muc is set?